test/forwarddiff_symbolic_dual_ops.jl

```diff
@@ -107,3 +107,10 @@ for f ∈ (hypot, muladd)
 end
 
 # fma is not defined for Symbolics.Num
+
+# https://github.com/JuliaSymbolics/Symbolics.jl/issues/1246
+@testset "isequal type ambiguity" begin
+    @variables x
+    xfunc(xval) = isequal(x, xval) ? xval : xval

```

Comment:
```suggestion
    xfunc(xval) = isequal(x, xval) ? x : xval
```

I don't quite understand this test? In both cases the derivative is respect to the value, so even if `isequal` is wrong this would pass. It needs to have a "bad" branch that would be wrong/error

Reply:
Thanks, I agree. I was very focused on simply ensuring it does not crash. I will improve it a bit and add a bad branch.